src/test/java/no/sikt/sws/IndexHandlerTest.java

```diff
@@ -145,25 +146,29 @@ void shuldPassQueryParams() throws IOException {
         assertThat(response.getStatusCode(), is(equalTo(HTTP_OK)));
     }
 
-    @Test
-    void shouldHandleSearchRequestWithQueryParameters() throws IOException {
+    @TestFactory
+    @DisplayName("Opensearch parameter requests")
+    public Stream<DynamicTest> testRequestWithQueryParameters() {
+
+        var responseTests = allRequestArguments().filter(TestCaseSws::isParamRequestTest);
+        return DynamicTest.stream(responseTests, TestCaseSws::getName, this::searchRequestWithQueryParameters);
+    }
 
-        var testcase =
-                new TestCaseLoader("proxy/requests-search.json")
-                .getTestCase("GET search by queryparameter");
+    void searchRequestWithQueryParameters(TestCaseSws testCase) throws IOException {
```

Comment:
We could have a better name here reflecting what it does (it asserts something?).
 Something like "assertTestCaseWithRequestQueryParameters" maybe?